cpp/arcticdb/version/test/test_version_map.cpp

```diff
@@ -547,56 +547,65 @@ TEST(VersionMap, StorageLogging) {
     ASSERT_EQ(tomb_keys, 3u);
 }
 
-std::shared_ptr<VersionMapEntry> write_two_versions(
-    std::shared_ptr<InMemoryStore> store,
-    std::shared_ptr<VersionMap> version_map,
-    const StreamId& id) {
+struct VersionChainOperation {
+    enum class Type {
+        WRITE,
+        TOMBSTONE,
+        TOMBSTONE_ALL
+    } type {Type::WRITE};
+
+    VersionId version_id { 0 };
+};
+
+std::shared_ptr<VersionMapEntry> write_versions(
+    const std::shared_ptr<InMemoryStore>& store,
+    const std::shared_ptr<VersionMap>& version_map,
+    const StreamId& id,
+    const std::vector<VersionChainOperation>& operations) {
     auto entry = version_map->check_reload(
             store,
             id,
             LoadStrategy{LoadType::NOT_LOADED, LoadObjective::INCLUDE_DELETED},
             __FUNCTION__);
 
-    auto key1 = atom_key_with_version(id, 0, 0);
-    version_map->do_write(store, key1, entry);
-    write_symbol_ref(store, key1, std::nullopt, entry->head_.value());
-    auto key2 = atom_key_with_version(id, 1, 1);
-    version_map->do_write(store, key2, entry);
-    // We override the symbol ref without a prev_key on purpose. This way we'll only load the version=1 from the ref key
-    write_symbol_ref(store, key2, std::nullopt, entry->head_.value());
+    for (const auto& [type, version_id]: operations) {
+        switch (type) {
+            case VersionChainOperation::Type::WRITE: {
+                auto key = atom_key_with_version(id, version_id, version_id);
+                version_map->do_write(store, key, entry);
+                write_symbol_ref(store, key, std::nullopt, entry->head_.value());
+                break;
+            }
+            case VersionChainOperation::Type::TOMBSTONE: {
+                version_map->write_tombstone(store, version_id, id, entry);
+                break;
+            }
+            case VersionChainOperation::Type::TOMBSTONE_ALL: {
+                version_map->delete_all_versions(store, id);
```

Comment:
Nit: Theoretically we can still have a tombstone_all to a version_id different than the last.
You can instead use the `write_tombstone_all_internal` + `write_symbol_ref` to do that.

If we decide against that it would be good to document that `VersionChainOperation{TOMBSTONE_ALL, 3}` might not do what we expect.